### PR TITLE
add unit tests for tags queries

### PR DESCRIPTION
**Pull request by the-mikedavis**
Following up on https://github.com/tree-sitter/tree-sitter/pull/1547, this PR adds some unit tests for the tags queries. Luckily these queries are pretty straightforward, so it only takes a few tests to get good coverage. If any edge cases come to mind that would be good to test, lmk!

Also included a commit to bump the cli to v0.20.6 which is the version where the tags tests was released.

### CODE REVIEW

File: src/tree_sitter/parser.h [hunk 1]
```diff
@@ -123,6 +123,7 @@ struct TSLanguage {
     unsigned (*serialize)(void *, char *);
     void (*deserialize)(void *, const char *, unsigned);
   } external_scanner;
+  const TSStateId *primary_state_ids;
```
Comment (amaanq) [inline src/tree_sitter/parser.h:126]: Hmm, does this stem from an older tree-sitter cli generation? If so, it'd be nice to see if it shows up generating now (and if not, remove it :) )
